activerecord/test/cases/insert_all_test.rb

```diff
@@ -104,6 +104,42 @@ def test_insert_all_can_skip_duplicate_records
     end
   end
 
+  def test_insert_all_with_skip_duplicates_and_autonumber_id_not_given
+    skip unless supports_insert_conflict_target?
+
+    assert_difference "Book.count", 1 do
+      # These two books are duplicates according to an index on %i[author_id name]
+      # but their IDs are not specified so they will be assigned different IDs
+      # by autonumber. We will get an exception from MySQL if we attempt to skip
+      # one of these records by assigning its ID.
+      Book.insert_all [
+        { author_id: 8, name: "Refactoring" },
+        { author_id: 8, name: "Refactoring" }
+      ]
+    end
+  end
+
+  def test_insert_all_with_skip_duplicates_and_autonumber_id_given
+    skip unless supports_insert_conflict_target?
+
+    assert_difference "Book.count", 1 do
+      Book.insert_all [
+        { id: 200, author_id: 8, name: "Refactoring" },
+        { id: 201, author_id: 8, name: "Refactoring" }
+      ]
+    end
+  end
+
+  def test_mysql_skip_duplicates_strategy_does_not_secretly_upsert
```

Comment:
Let's remove `mysql_`.

Reply:
Also `skip unless supports_insert_on_duplicate_skip?` is needed.

https://travis-ci.org/rails/rails/jobs/515997440#L1214

Reply:
No problem! Change made.

I just wanted to confirm, though — these tests all specifically cover MySQL's strategy for skipping duplicates — its strategy of attempting a no-op upsert.

And this one — which asserts that  "skip_duplicates_strategy_does_not_secretly_upsert" — is kind of a confusing thing to assert of the other connection adapters.

Reply:
Yeah, whatever the strategy we should test just behavior, better naming is welcome.